sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -488,6 +542,14 @@ long flush(
               return RetryType.RETRY_ALL_OPERATIONS;
             },
             c -> {
+              AppendRowsResponse response = Preconditions.checkStateNotNull(c.getResult());
+              if (autoUpdateSchema && response.hasUpdatedSchema()) {
```

Comment:
do we get a similar information piece on schema updates when the insert fails? if so that info added to the failed insert rows could help on debugging.

Reply:
I don't believe we do.